ibm/resource_ibm_private_dns_glb_pool.go

```diff
@@ -0,0 +1,340 @@
+
+	getPoolOptions := sess.NewGetPoolOptions(idset[0], idset[1])
+	presponse, resp, err := sess.GetPool(getPoolOptions)
+	if err != nil {
+		return fmt.Errorf("Error fetching pdns GLB Pool:%s\n%s", err, resp)
+	}
+
+	response := *presponse
+	d.Set("id", response.ID)
+	d.Set(pdnsGlbPoolName, response.Name)
+	d.Set(pdnsGlbPoolID, response.ID)
+	d.Set(pdnsGlbPoolDescription, response.Description)
+	d.Set(pdnsGlbPoolEnabled, response.Enabled)
+	d.Set(pdnsGlbPoolHealthyOriginsThreshold, response.HealthyOriginsThreshold)
+	d.Set(pdnsGlbPoolMonitor, response.Monitor)
+	d.Set(pdnsGlbPoolChannel, response.NotificationChannel)
+	d.Set(pdnsGlbPoolRegion, response.HealthcheckRegion)
+	d.Set(pdnsGlbPoolSubnet, response.HealthcheckSubnets)
+	d.Set(pdnsGlbPoolCreatedOn, response.CreatedOn)
+	d.Set(pdnsGlbPoolModifiedOn, response.ModifiedOn)
+	d.Set(pdnsGlbPoolOrigins, flattenPoolOrigins(response.Origins))
+
+	return nil
+}
+
+func flattenPoolOrigins(list []dns.Origin) []map[string]interface{} {
+	origins := []map[string]interface{}{}
+	for _, origin := range list {
+		l := map[string]interface{}{
+			pdnsGlbPoolOriginsName:        *origin.Name,
+			pdnsGlbPoolOriginsAddress:     *origin.Address,
+			pdnsGlbPoolOriginsEnabled:     *origin.Enabled,
+			pdnsGlbPoolOriginsDescription: *origin.Description,
+		}
+		origins = append(origins, l)
+	}
+	return origins
+}
+
+func resourceIBMPrivateDNSGLBPoolUpdate(d *schema.ResourceData, meta interface{}) error {
+	sess, err := meta.(ClientSession).PrivateDnsClientSession()
+	if err != nil {
+		return err
+	}
+
+	idset := strings.Split(d.Id(), "/")
+	updatePoolOptions := sess.NewUpdatePoolOptions(idset[0], idset[1])
+
+	if d.HasChange(pdnsGlbPoolName) ||
+		d.HasChange(pdnsGlbPoolDescription) ||
+		d.HasChange(pdnsGlbPoolEnabled) ||
+		d.HasChange(pdnsGlbPoolHealthyOriginsThreshold) ||
+		d.HasChange(pdnsGlbPoolMonitor) ||
+		d.HasChange(pdnsGlbPoolChannel) ||
+		d.HasChange(pdnsGlbPoolRegion) ||
+		d.HasChange(pdnsGlbPoolOrigins) ||
+		d.HasChange(pdnsGlbPoolSubnet) {
+		if Mname, ok := d.GetOk(pdnsGlbPoolName); ok {
+			updatePoolOptions.SetName(Mname.(string))
+		}
+		if description, ok := d.GetOk(pdnsGlbPoolDescription); ok {
+			updatePoolOptions.SetDescription(description.(string))
+		}
+		if enable, ok := d.GetOk(pdnsGlbPoolEnabled); ok {
+			updatePoolOptions.SetEnabled(enable.(bool))
+		}
+		if threshold, ok := d.GetOk(pdnsGlbPoolHealthyOriginsThreshold); ok {
+			updatePoolOptions.SetHealthyOriginsThreshold(int64(threshold.(int)))
+		}
+		if monitor, ok := d.GetOk(pdnsGlbPoolMonitor); ok {
+			updatePoolOptions.SetMonitor(monitor.(string))
+		}
+		if chanel, ok := d.GetOk(pdnsGlbPoolChannel); ok {
+			updatePoolOptions.SetNotificationChannel(chanel.(string))
+		}
+		if region, ok := d.GetOk(pdnsGlbPoolRegion); ok {
+			updatePoolOptions.SetHealthcheckRegion(region.(string))
+		}
+		if subnet, ok := d.GetOk(pdnsGlbPoolSubnet); ok {
+			updatePoolOptions.SetHealthcheckSubnets(subnet.([]string))
```

Comment:
use expandStringList() method to take the subnet. do the same in create()

Reply:
Done